featurebyte/common/utils.py

```diff
@@ -471,3 +474,29 @@ def is_server_mode() -> bool:
     """
     sdk_execution_mode = os.environ.get("FEATUREBYTE_SDK_EXECUTION_MODE")
     return sdk_execution_mode == "SERVER"
+
+
+@contextmanager
+def timer(message: str, logger: logging.Logger) -> Generator[None, None, None]:
+    """
+    Timer context manager to measure execution time.
+
+    Parameters
+    ----------
+    message: str
+        Message to log before and after the execution time measurement.
+    logger: logging.Logger
+        Logger object to log the execution time message.
+
+    Yields
+    ------
+    Generator[None, None, None]
+        A context manager that logs the execution time with the given message and logger.
+    """
+    start_time = time.time()
+    try:
+        yield
+    finally:
+        end_time = time.time()
+        duration = end_time - start_time
+        logger.info(f"{message}: {duration} seconds")
```

Comment:
Might be useful if this utility allows specifying extra dict that will be passed to logger.